core/server/worker/src/main/java/alluxio/worker/block/meta/StorageTierView.java

```diff
@@ -41,7 +42,7 @@ public StorageTierView(StorageTier tier) {
    * @return a list of directory views in this storage tier view
    */
   public List<StorageDirView> getDirViews() {
-    return Collections.unmodifiableList(mDirViews);
+    return new ArrayList<>(mDirViews.values());
```

Comment:
Does this List need to be modifiable now?

Reply:
No. it's only changed because we now need to convert the value set of the map to a list. It is equivalent to the old unmodifiable list in the sense that user of this API won't be able to change the underlying directory collection stored in this view.

Reply:
but the user of this API will not get an error if they try to modify this list. In fact they will successfully modify the returned list right?

Reply:
yeah I think the original purpose of wrapping the collection with `unmodifiableList` is to prevent mutation of the view rather than prevent the mutation of the return value after method returns. I don't feel it is necessary to add an unmodifiable wrapper on top of the list now that we already create a copy of the collection. Let me know if you think there is a strong reason for an additional wrapper otherwise.

Reply:
sounds good